crates/metallic/src/target.rs

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+
+use std::fs;
+
+pub struct Target;
+
+const CACHEDIR_CONTENTS: &str = r#"Signature: 8a477f597d28d172789f06886806bc55
+# This file is a cache directory tag created by metallic.
+# For information about cache directory tags, see:
+#	http://www.brynosaurus.com/cachedir/"#;
+
+const GITIGNORE_CONTENTS: &str = "*";
+
+impl Target {
+    pub fn setup() {
+        // ensure the directories all exist
+        fs::create_dir_all("./target/ice").unwrap();
+        fs::create_dir_all("./target/.llvm").unwrap();
```

Comment:
I was mostly being lazy with the naming here because `.llvm` will be temporary. Binaries would only be cached as assembly so the LLVM IR building here is temporary because building to assembly is out of scope for this PR.

Reply:
This may still be useful for debugging in the future, so i'd rather change it